Fix typos and errors in API Definition v1.0

stevenwjy opened the pull request:
Hi everyone, this is a draft pull request to fix typos and errors in the API Definition v1.0 that I encountered when reading the specification. Hope this could help to make the documentation better!

Besides, I also wrote some comments as I read through the document, which I try to list down [here](https://docs.google.com/document/d/1jYVCQWUvRV8sMZRtc43g1v9ls_Oz7n-qAxPjHB4UaeA/edit?usp=sharing). Thank you @lewisdaly for answering a lot of questions that I had, which probably some have been removed as they are no longer relevant for potential improvements. Hope that my write-ups can be used to improve some other parts of the current specification.

I wrote the comments based on my perspective as someone new to Mojaloop. Some of them are also based the challenges that I had when trying to understand how things work here. Hence, sorry if there are a lot of unimportant suggestions/questions. Feel free to comment out your thoughts on it and I will try to keep the document updated by removing some answered questions or rejected suggestions.

Thank you!

elnyry-sam-k commented on assets/diagrams/sequence/figure65.plantuml:
```diff
@@ -317,24 +317,24 @@ else
     end hnote
     autonumber resume
-    PayerFSP ->> Switch: **GET /transactions/**<i><TransactionID></i>
+    PayerFSP ->> Switch: **GET /transactions/**<i><ID></i>
```
Please leave it as it is, this is by design

stevenwjy replied:
Okay, got it, reverted all similar changes.

Besides, in my new commit, I changed `PUT /participants/<ID>` to `PUT /participants/<TransactionID> in order to be consistent. Do you think it is okay or perhaps should I revert that again?

elnyry-sam-k replied:
Thanks @stevenwjy. For /participants it will be <ID> , please revert if needed

stevenwjy replied:
Sorry, I don't get what you mean, should I revert it back to `PUT /participants/<ID>`?

elnyry-sam-k replied:
Yes, please

stevenwjy replied:
Oh sorry... just realised that what I mean was `PUT /transactions/...`

However, when I was about to revert it, I realised that it would contradict with your other comments to change `PUT /transactions/<ID>` to `PUT /transactions/<TransactionID>` in figure 63. 

What I meant before was that I changed `PUT /transactions/<ID>` to `PUT /transactions/<TransactionID>` in figure 56, 64, and 65. Are you sure that you want to differentiate between figure 63 and the others?

If it is intended then I will revert my changes in figure 56, 64, and 65.

elnyry-sam-k replied:
Ok, if you changed it to `PUT /transactions/<TransactionID>`, then we should be fine..

elnyry-sam-k commented on assets/diagrams/sequence/figure65.plantuml:
```diff
@@ -317,24 +317,24 @@ else
     PayerFSP <<-- Switch: **HTTP 202** (Accepted)
     autonumber resume
-    Switch ->> PayeeFSP: **GET /transactions/**<i><TransactionID></i>
+    Switch ->> PayeeFSP: **GET /transactions/**<i><ID></i>
```
Please leave it as it is, this is by design

elnyry-sam-k commented on assets/diagrams/sequence/figure65.plantuml:
```diff
@@ -237,12 +237,12 @@ Payee -> Payee: Payer\napproves/rejects\ntransaction\nusing OTP
 deactivate PayeeFSP
 autonumber resume
-PayerFSP <<- Switch: **PUT /authorizations/**\n<i><TransactionRequestID></i>\n(OTP)
+PayerFSP <<- Switch: **PUT /authorizations/**<i><ID></i>\n(OTP)
```
Please leave it as it is, this is by design

elnyry-sam-k commented on assets/diagrams/sequence/figure65.plantuml:
```diff
@@ -237,12 +237,12 @@ Payee -> Payee: Payer\napproves/rejects\ntransaction\nusing OTP
 PayeeFSP <<-- Payee: Entered OTP
 autonumber resume
-Switch <<- PayeeFSP: **PUT /authorizations/**<i><TransactionRequestID></i>\n(OTP)
+Switch <<- PayeeFSP: **PUT /authorizations/**<i><ID></i>\n(OTP)
```
Please leave it as it is, this is by design

elnyry-sam-k commented on assets/diagrams/sequence/figure65.plantuml:
```diff
@@ -222,12 +222,12 @@ Group #Oldlace Option #2
   Payer <- PayerFSP: Use OTP\nto confirm
 end
-PayerFSP ->> Switch: **GET /authorizations/**\n<i><TransactionRequestID></i>\n(Amount and fees)
+PayerFSP ->> Switch: **GET /authorizations/**<i><ID></i>\n(Amount and fees)
```
Please leave it as it is, this is by design

elnyry-sam-k commented on assets/diagrams/sequence/figure65.plantuml:
```diff
@@ -222,12 +222,12 @@ Group #Oldlace Option #2
 PayerFSP <<-- Switch: **HTTP 202** (Accepted)
 autonumber resume
-Switch ->> PayeeFSP: **GET /authorizations/**\n<i><TransactionRequestID></i>\n(Amount and fees)
+Switch ->> PayeeFSP: **GET /authorizations/**<i><ID></i>\n(Amount and fees)
```
Please leave it as it is, this is by design

elnyry-sam-k commented on assets/diagrams/sequence/figure64.plantuml:
```diff
@@ -191,11 +191,11 @@ PayerFSP -->> Switch: **HTTP 200** (OK)
 autonumber resume
 PayerFSP -> PayerFSP: Rate Payer FSP quote\n(depending on the fee model)
-Payer <- PayerFSP: Will you approve\ntransaction\nrequest for\n100 USD(plusfees)\nto Payee?
+Payer <- PayerFSP: Will you approve the\ntransaction request\nfor 100 USD\n(including fees)\nto Payee?
```
Please retain the "plus fees" phrase

elnyry-sam-k commented on assets/diagrams/sequence/figure64.plantuml:
```diff
@@ -255,23 +255,23 @@ else
     end hnote
     autonumber resume
-    PayerFSP ->> Switch: **GET /transactions/**<i><TransactionID></i>
+    PayerFSP ->> Switch: **GET /transactions/**<i><ID></i>
```
Please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure64.plantuml:
```diff
@@ -255,23 +255,23 @@ else
     PayerFSP <<-- Switch: **HTTP 202** (Accepted)
     autonumber resume
-    Switch ->> PayeeFSP: **GET /transactions/**<i><TransactionID></i>
+    Switch ->> PayeeFSP: **GET /transactions/**<i><ID></i>
```
Please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure64.plantuml:
```diff
@@ -191,11 +191,11 @@ PayerFSP -->> Switch: **HTTP 200** (OK)
 Alt #OldLace Alternatives
   hnote over of Payer #OldLace
-    User rejects 
+    User rejects the
```
Please leave it as it is - lets only fix mistakes

elnyry-sam-k commented on assets/diagrams/sequence/figure64.plantuml:
```diff
@@ -218,7 +218,7 @@ Alt #OldLace Alternatives
 else
   hnote over of Payer #OldLace
-    User rejects 
+    User approves the
```
Thanks for catching this

elnyry-sam-k commented on assets/diagrams/sequence/figure63.plantuml:
```diff
@@ -230,23 +230,23 @@ group #OldLace Optional
     Get transaction data
   end hnote
-  PayerFSP ->> Switch: **GET /transactions/**<i><TransactionID></i>
+  PayerFSP ->> Switch: **GET /transactions/**<i><ID></i>
```
again, please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure63.plantuml:
```diff
@@ -230,23 +230,23 @@ group #OldLace Optional
   PayerFSP <<-- Switch: **HTTP 202** (Accepted)
   autonumber resume
-  Switch <<- PayeeFSP: **GET /transactions/**<i><TransactionID></i>
+  Switch <<- PayeeFSP: **GET /transactions/**<i><ID></i>
```
please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure63.plantuml:
```diff
@@ -230,23 +230,23 @@ group #OldLace Optional
   autonumber resume
   PayeeFSP -> PayeeFSP: Lookup\ntransaction\ninformation
-  Switch <<- PayeeFSP: **PUT /transactions/**<i><TransactionID></i>\n(Transaction details)
+  Switch <<- PayeeFSP: **PUT /transactions/**<i><ID></i>\n(Transaction details)
```
please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure63.plantuml:
```diff
@@ -230,23 +230,23 @@ group #OldLace Optional
   deactivate PayeeFSP
   autonumber resume
-  PayerFSP <<- Switch: **PUT /transactions/**<i><TransactionID></i>\n(Transaction details)
+  PayerFSP <<- Switch: **PUT /transactions/**<i><ID></i>\n(Transaction details)
```
please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure56.plantuml:
```diff
@@ -120,10 +120,10 @@ activate PayerFSP
 PayerFSP -> PayerFSP: Perform quote\n(process not shown here)
 PayerFSP -> PayerFSP: Perform transfer\n(process not shown here)
-PayerFSP ->> Switch: **GET /transactions/**<i><transactionID></i>
+PayerFSP ->> Switch: **GET /transactions/**<i><ID></i>
```
please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure56.plantuml:
```diff
@@ -120,10 +120,10 @@ activate PayerFSP
 activate Switch
 PayerFSP <<-- Switch: **HTTP 202** (Accepted)
-Switch ->> PayeeFSP: **GET /transactions/**<i><transactionID></i>
+Switch ->> PayeeFSP: **GET /transactions/**<i><ID></i>
```
please leave it as it is

elnyry-sam-k commented on assets/diagrams/sequence/figure32.plantuml:
```diff
@@ -142,10 +142,10 @@ Switch ->> PayeeFSP: **POST /quotes**\n(amountType=RECEIVE,\namount=100 USD)
 Switch <<-- PayeeFSP: **HTTP 202** (Accepted)
 PayeeFSP -> PayeeFSP: Fee is 0 USD
-Switch <<- PayeeFSP: **PUT /quotes/**<i><ID></i>\n(transferAmount=100 USD,\npayeeFSPFee=2 USD)
+Switch <<- PayeeFSP: **PUT /quotes/**<i><ID></i>\n(transferAmount=100 USD,\npayeeFSPFee=0 USD)
```
```suggestion
Switch <<- PayeeFSP: **PUT /quotes/**<i><ID></i>\n(transferAmount=100 USD)
```

elnyry-sam-k commented on assets/diagrams/sequence/figure32.plantuml:
```diff
@@ -142,10 +142,10 @@ Switch ->> PayeeFSP: **POST /quotes**\n(amountType=RECEIVE,\namount=100 USD)
 Switch -->> PayeeFSP: **HTTP 200** (OK)
 deactivate PayeeFSP
-Switch ->> PayerFSP: **PUT /quotes/**<i><ID></i>\n(transferAmount=100 USD,\npayeeFSPFee=2 USD)
+Switch ->> PayerFSP: **PUT /quotes/**<i><ID></i>\n(transferAmount=100 USD,\npayeeFSPFee=0 USD)
```
```suggestion
Switch ->> PayerFSP: **PUT /quotes/**<i><ID></i>\n(transferAmount=100 USD)
```

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -238,11 +238,11 @@ The following HTTP methods, as defined in RFC 7231<sup>18</sup>, are supported b
   - Object information concerning a previously created object (HTTP **POST**) or sent information request (HTTP **GET**).
   - Acknowledgement that whether an object was deleted (HTTP **DELETE**).
-  - Error information in case the HTTP **POST** or HTTP **GET** request failed to be processed on the server.
+  - Error information in case the HTTP **GET**, **POST**, or **DELETE** request failed to be processed on the server.
```
Please revert this

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -361,7 +361,7 @@ To simplify duplicate analysis, it is recommended to create and store a hash val
 ### 3.3 API Versioning
 
-The strategy of the development of the API is to maintain backwards compatibility between the API and its resources and services to the maximum extent possible; however, changes to the API should be expected by implementing parties. Versioning of the API is specific to the API resource (for example, **/participants**, **/quotes**, **/transfers**).
+The API development strategy is to maintain backwards compatibility between the API and its resources and services to the maximum extent possible; however, changes to the API should be expected by implementing parties. Versioning of the API is specific to the API resource (for example, **/participants**, **/quotes**, **/transfers**).
```
Please revert this

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -564,7 +563,7 @@ To support conditional transfers for ILP, a ledger must support a transfer API t
 When the fulfilment of a transfer is submitted to a ledger, the ledger must ensure that the fulfilment is valid for the condition that was attached to the original transfer request. If it is valid, the transfer is committed, otherwise it is rejected, and the transfer remains in a pending state until a valid fulfilment is submitted or the transfer expires.
 
-ILP supports a variety of conditions for performing a conditional payment, but implementers of the API should use the SHA-256 hash of a 32-byte pre-image. The condition attached to the transfer is the SHA-256 hash and the fulfilment of that condition is the pre-image. Therefore, if the condition attached to a transfer is a SHA -256 hash, then when a fulfilment is submitted for that transaction, t ...
+ILP supports a variety of conditions for performing a conditional payment, but implementers of the API should use the SHA-256 hash of a 32-byte preimage. The condition attached to the transfer is the SHA-256 hash and the fulfilment of that condition is the preimage. Therefore, if the condition attached to a transfer is a SHA-256 hash, then when a fulfilment is submitted for that transaction, the  ...
```
please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -833,7 +834,7 @@ Payee Receive Amount = Transfer Amount - Payee FSP Fee + Payee FSP Commission
 **Listing 11 -- Relation between transfer amount and Payee receive amount**
 
-The actual Payee receive amount including any internal Payee FSP fees can optionally be sent by the Payee FSP to the Payer FSP in the Quote callback, see element **payeeReceiveAmount** in [Table 18](#table-18).
+The actual amount that Payee receives including any internal Payee FSP fees can optionally be sent by the Payee FSP to the Payer FSP in the Quote callback, see element **payeeReceiveAmount** in [Table 18](#table-18).
```
Please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -845,7 +846,7 @@ Tax on Agent Commission is tax for an _Agent_ as a result of the Agent receiving
 #### 5.1.5.2 Tax on FSP Internal Fee
 
-FSPs could be taxed on FSP internal fees that they receive from the transactions; for example, Payer fees to Payer FSP or Payee fees to Payee FSP. This tax should be handled internally within the FSP and collected by the FSPs because they receive a fee.
+FSPs could be taxed on FSP internal fees that they receive from the transactions; for example, Payer fees to Payer FSP or Payee fees to Payee FSP. This tax should be handled internally within the FSPs and collected by the FSPs because they receive the fees.
```
Please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -861,11 +862,11 @@ In the API, there is a possibility for a Payee FSP to add a commission to either
 #### 5.1.5.5.1 Non-Disclosing of Fees
 
-For non-disclosing of fees, all FSP commission from the Payee FSP should understood as the Payer FSP receiving a fee from the Payee FSP. The tax on the received fee should be handled internally within the Payer FSP, similar to the way it is handled in [Section 5.1.5.2](#5152-tax-on-fsp-internal-fee).
+For non-disclosing of fees, all FSP commission from the Payee FSP must be considered as the Payer FSP receiving a fee from the Payee FSP. The tax on the received fee should be handled internally within the Payer FSP, similar to the way it is handled in [Section 5.1.5.2](#5152-tax-on-fsp-internal-fee).
```
Please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -861,11 +862,11 @@ In the API, there is a possibility for a Payee FSP to add a commission to either
 #### 5.1.5.5.2 Disclosing of Fees
 
-If the Payee FSP commission amount is less than or equal to the amount of transaction fees originating from the Payer FSP, then the Payee FSP commission should always be understood as being used for covering fees that the Payer would otherwise need to pay.
+If the Payee FSP commission amount is less than or equal to the amount of transaction fees originating from the Payer FSP, then the Payee FSP commission must be considered as being used for covering fees that the Payer would otherwise need to pay.
```
Please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -1109,7 +1107,7 @@ See [Figure 39](#figure-39) for a highly simplified view of the movement of mone
 ### 5.2 Party Addressing
 
-Both Parties in a financial transaction, (that is, the Payer and the Payee) are addressed in the API by a _Party ID Type_ (element **PartyIdType**, [Section 7.3.25](#7325-partyidtype)), **a** _Party ID_ (**PartyIdentifier**, [Section 7.3.24](#7324-partyidentifier)), **and** an optional _Party Sub ID or Type_ (**PartySubIdOrType**, [Section 7.3.27](#7327-partysubidortype)). Some Sub-Types are pre- ...
+Both Parties in a financial transaction, (that is, the Payer and the Payee) are addressed in the API by a _Party ID Type_ (**PartyIdType**, [Section 7.3.25](#7325-partyidtype)), a _Party ID_ (**PartyIdentifier**, [Section 7.3.24](#7324-partyidentifier)), and an optional _Party Sub ID or Type_ (**PartySubIdOrType**, [Section 7.3.27](#7327-partysubidortype)). Some Sub-Types are pre-defined in the A ...
```
```suggestion
Both Parties in a financial transaction, (that is, the Payer and the Payee) are addressed in the API by a _Party ID Type_ (element **PartyIdType**, [Section 7.3.25](#7325-partyidtype)), a _Party ID_ (**PartyIdentifier**, [Section 7.3.24](#7324-partyidentifier)), and an optional _Party Sub ID or Type_ (**PartySubIdOrType**, [Section 7.3.27](#7327-partysubidortype)). Some Sub-Types are pre-defined in the API for personal identifiers (**PersonalIdentifierType**, [Section 7.5.7](#757-personalidentifiertype)); for example, for passport number or driver's license number.
```

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -1752,7 +1750,7 @@ For more information regarding Quoting, see [Section 5.1](#51-quoting).
 {% enduml %}
 
-**Figure 46 -- Example process for resource /quotes**
+**Figure 46 -- Example process for the API resource /quotes**
```
Please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -2089,7 +2087,7 @@ In [Figure 51](#figure-51), an expiry has been set to 30 seconds from the curren
 **Figure 52 -- Client receiving an expired transfer**
 
-To limit these kinds of error scenarios, the clients (Payer FSP and optional Switch in [Figure 52](#figure-52)) participating in the ILP transfer should allow some extra time after actual expiry time during which the callback from the server can be received. The client or clients should also query the server after expiry, but before the end of the extra time, if any callback from the server has b ...
+To limit these kinds of error scenarios, the clients (Payer FSP and optional Switch in [Figure 52](#figure-52)) participating in the ILP transfer should allow some extra time after actual expiry time during which the callback from the server can be received. The client(s) should also query the server after expiry, but before the end of the extra time, if any callback from the server has been lost ...
```
Please revert

elnyry-sam-k commented on documents/API Definition v1.0.md:
```diff
@@ -2249,11 +2247,11 @@ This section defines the logical API resource **Transactions**, described in _Ge
 #### 6.8.1 Service Details
 
-[Figure 56](#figure-56) shows an example for the transaction process. The actual transaction will be performed as part of the transfer process. The service **GET /transactions/**_{TransactionID}_ can then be used to get more information about the financial transaction that was performed as part of the transfer process.
+[Figure 56](#figure-56) shows an example for the transaction process. The actual transaction will be performed as part of the transfer process. The service **GET /transactions/**_{ID}_ can then be used to get more information about the financial transaction that was performed as part of the transfer process.
```
Please revert

elnyry-sam-k commented:
@stevenwjy - I reviewed this and made comments. Please make those changes to be able to merge this.

Thanks for doing this - it was very comprehensive. I would encourage though in future to keep PRs manageable by keeping changes to a limited count and use separate PRs for more changes to make reviews and editing easier.

Thanks again!

stevenwjy commented:
Thank you @elnyry-sam-k for reviewing this PR! Sorry for taking some time to update this...

I have reverted some changes based on your review, but I also made a minor change regarding `<ID>` vs `<TransactionID>` as I reply to one of your comments. Please let me know if it is also by design and I need to revert my changes again.

Besides, as I mentioned before, I have updated the description for this pull request to put down the link for my comments on the current API specification. It would be great if you could help to take a look and leave some of your thoughts there. Also, please let me know if I could help to make any other updates in this PR or perhaps using another PR.

elnyry-sam-k commented:
Thanks a lot @stevenwjy .. Yes please finalize that single change and I think we should be good..

This is greatly helpful!

elnyry-sam-k commented:
@stevenwjy - please let me know when its ready.. I think we addressed the last open conversation

stevenwjy commented:
> @stevenwjy - please let me know when its ready.. I think we addressed the last open conversation

Yup, I think the PR is ready 👍 
Thank you for the review @elnyry-sam-k !!

stevenwjy commented:
Btw @elnyry-sam-k, I think there are also a few other issues in some svg files that I mentioned in my comments (the link is in the description of this PR). I haven't had the chance to fix it but I think it could be addressed using another PR if you are okay with it.

elnyry-sam-k commented:
@stevenwjy Thank you! Yes, lets have separate PRs for any new issues in SVGs or otherwise :-)
